cmd/flag/proxy_flags.go

```diff
@@ -30,38 +30,50 @@ var (
 	defaultProxyProtocol = "udp"
 
 	// Supported proxy.
-	availableProxy = []string{"udp"}
-	// Default listen host
-	defaultListenHost = "127.0.0.1"
-	// Default listen port
-	defaultListenPort uint16 = 6000
-	// Default send host
-	defaultSendHost = "127.0.0.1"
-	// Default send port
-	defaultSendPort uint16 = 6001
+	availableProxy = []string{"udp", "tcp"}
+	// Default listen host for UDP.
+	defaultUDPListenHost = "127.0.0.1"
+	// Default listen port for UDP.
+	defaultUDPListenPort uint16 = 6000
+	// Default send host for UDP.
+	defaultUDPSendHost = "127.0.0.1"
+	// Default send port for UDP.
+	defaultUDPSendPort uint16 = 6001
+
+	// Default listen host for TCP.
+	defaultTCPListenHost = "127.0.0.1"
+	// Default listen port for TCP.
+	defaultTCPListenPort uint16 = 4000
 )
 
 type ProxyFlags struct {
 	ProxyProtocol string
 
-	ListenHost string
-	ListenPort uint16
-	SendHost   string
-	SendPort   uint16
+	UDPListenHost string
+	UDPListenPort uint16
+	UDPSendHost   string
+	UDPSendPort   uint16
+
+	TCPListenHost string
+	TCPListenPort uint16
 }
 
 // Add flags to the command.
 func (f *ProxyFlags) AddFlags(cmd *cobra.Command) {
 	cmd.Flags().StringVarP(&f.ProxyProtocol, "proxy", "", defaultProxyProtocol,
-		"Proxy protocol. One of: "+strings.Join(availableFormats, "|"))
-	cmd.Flags().StringVar(&f.ListenHost, "listen-host", defaultListenHost,
+		"Proxy protocol. One of: "+strings.Join(availableProxy, "|"))
+	cmd.Flags().StringVar(&f.UDPListenHost, "udp-listen-host", defaultUDPListenHost,
 		"The host to listen for packets on.")
-	cmd.Flags().Uint16Var(&f.ListenPort, "listen-port", defaultListenPort,
+	cmd.Flags().Uint16Var(&f.UDPListenPort, "udp-listen-port", defaultUDPListenPort,
 		"The port stellar listens for packets on. Packets on this port will be sent to the satellite.")
-	cmd.Flags().StringVar(&f.SendHost, "send-host", defaultSendHost,
```

Comment:
Can we keep the existing flats too, document they are for UDP, and add that they are deprecated?

Reply:
Added current flags and mentioned them as deprecated. Values set to old flags are set to new flags.